runtime/compiler/aarch64/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -24,14 +24,71 @@
 #include "codegen/CodeGenerator.hpp"
 
 J9::ARM64::AheadOfTimeCompile::AheadOfTimeCompile(TR::CodeGenerator *cg) :
-         J9::AheadOfTimeCompile(_relocationTargetTypeToHeaderSizeMap, cg->comp()),
-         _cg(cg)
+      J9::AheadOfTimeCompile(_relocationTargetTypeToHeaderSizeMap, cg->comp()),
+      _cg(cg),
+      _relocationList(getTypedAllocator<TR::ARM64Relocation*>(cg->comp()->allocator()))
    {
    }
 
 void J9::ARM64::AheadOfTimeCompile::processRelocations()
    {
-   TR_UNIMPLEMENTED();
+   TR_J9VMBase *fej9 = (TR_J9VMBase *)(_cg->fe());
+   TR::IteratedExternalRelocation *r;
+
+   for (auto iterator = getRelocationList().begin(); iterator != getRelocationList().end(); ++iterator)
+      (*iterator)->mapRelocation(_cg);
```

Comment:
This whole thing is extremely convoluted. The `_relocationList` seems to only be really used on POWER, where they use it as a variant of the `TR::BeforeBinaryEncodingExternalRelocation` external relocation; whereas the latter only has one `TR::Instruction`, the former has two - the relo infra needs to (after binary encoding) translate the `TR::Instruction` into a real address. On POWER, `_relocationList` is used when converting `TR::PPCPairedRelocation` to `TR::ExternalOrderedPair32BitRelocation` via `mapRelocation`.

x86 does not have a `mapRelocation` defined. Z though does. However, Z doesn't ever populate `_relocationList`; it seems like it was either just copied over from POWER without understanding what the point was, or Z supported Ordered Pair 32bit relocations.

The point of all this is: `mapRelocation` is a pure virtual function, so there's going to need to be an implementation for it on AArch64. However, I don't know if AArch64 even needs this. 

Perhaps the conservative thing to do is to provide the implementation for `mapRelocation`. However, the fact that `_relocationList` is only used to create `TR::ExternalOrderedPair32BitRelocation`s seems arbitrary, which is why if you're gonna choose to provide an implemention, I'd prefer the AArch64 version of `mapRelocation` to just be empty; it's better than something that seems to have a generic name (eg `_relocationlist`) imply something specific (eg OrderedPair32Bit).

Reply:
I stopped adding `_relocationList` and stopped calling `(*iterator)->mapRelocation(_cg)` on elements in the list.
Nobody is adding elements to the `_relocationList` on AArch64 at this point.